MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2581,38 +2628,64 @@ public string GetHardwareAcceleratedVideoDecoder(VideoType videoType, MediaStrea
                         case "h264":
                             if (_mediaEncoder.SupportsDecoder("h264_cuvid") && encodingOptions.HardwareDecodingCodecs.Contains("h264", StringComparer.OrdinalIgnoreCase))
                             {
-                                // cuvid decoder does not support 10-bit input
-                                if ((videoStream.BitDepth ?? 8) > 8)
-                                {
-                                    encodingOptions.HardwareDecodingCodecs = Array.Empty<string>();
```

Comment:
so this might be a source of a bug when HW resets itself upon playing some incompatible media... cc @dkanada